pkg/operation/botanist/extensions/infrastructure/infrastructure_test.go

```diff
@@ -0,0 +1,304 @@
+			Region:         region,
+		}
+		infra = &extensionsv1alpha1.Infrastructure{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
+		deployWaiter = infrastructure.New(log, c, values, time.Millisecond, 2*time.Millisecond, 3*time.Millisecond)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("#Deploy", func() {
+		DescribeTable("correct Infrastructure is created", func(mutator func()) {
+			defer test.WithVars(
+				&infrastructure.TimeNow, mockNow.Do,
+			)()
+			mockNow.EXPECT().Do().Return(now.UTC()).AnyTimes()
+
+			c.EXPECT().Get(ctx, kutil.Key(namespace, name), infra.DeepCopy()).DoAndReturn(func(_ context.Context, _ client.ObjectKey, obj *extensionsv1alpha1.Infrastructure) error {
+				return apierrors.NewNotFound(schema.GroupResource{}, name)
+			})
+
+			deployWaiter.SetSSHPublicKey(sshPublicKey)
+			infra.Spec = extensionsv1alpha1.InfrastructureSpec{
+				DefaultSpec: extensionsv1alpha1.DefaultSpec{
+					Type:           providerType,
+					ProviderConfig: providerConfig,
+				},
+				Region:       region,
+				SSHPublicKey: sshPublicKey,
+				SecretRef: corev1.SecretReference{
+					Name:      v1beta1constants.SecretNameCloudProvider,
+					Namespace: namespace,
+				},
+			}
+			mutator()
+			c.EXPECT().Create(ctx, infra)
+
+			Expect(deployWaiter.Deploy(ctx)).To(Succeed())
+		},
+			Entry("with no modification", func() {}),
+			Entry("without provider config", func() {
+				values.ProviderConfig = nil
+				infra.Spec.ProviderConfig = nil
+			}),
+			Entry("creation phase", func() {
+				values.IsInCreationPhase = true
+				infra.Annotations = map[string]string{
+					v1beta1constants.GardenerOperation: v1beta1constants.GardenerOperationReconcile,
+					v1beta1constants.GardenerTimestamp: now.UTC().String(),
+				}
+			}),
+			Entry("wake up phase", func() {
+				values.IsWakingUp = true
+				infra.Annotations = map[string]string{
+					v1beta1constants.GardenerOperation: v1beta1constants.GardenerOperationReconcile,
+					v1beta1constants.GardenerTimestamp: now.UTC().String(),
+				}
+			}),
+			Entry("deployment task", func() {
+				values.DeploymentRequested = true
+				infra.Annotations = map[string]string{
+					v1beta1constants.GardenerOperation: v1beta1constants.GardenerOperationReconcile,
+					v1beta1constants.GardenerTimestamp: now.UTC().String(),
+				}
+			}),
+			Entry("restoration phase", func() {
+				values.IsInRestorePhaseOfControlPlaneMigration = true
+				infra.Annotations = map[string]string{
+					v1beta1constants.GardenerOperation: v1beta1constants.GardenerOperationWaitForState,
+					v1beta1constants.GardenerTimestamp: now.UTC().String(),
+				}
+			}),
+		)
+	})
+
+	Describe("#Wait", func() {
+		It("should return error when it's not found", func() {
+			c.EXPECT().Get(gomock.Any(), kutil.Key(namespace, name), &extensionsv1alpha1.Infrastructure{}).DoAndReturn(func(_ context.Context, _ client.ObjectKey, obj *extensionsv1alpha1.Infrastructure) error {
```

Comment:
You could simply use `.Return` in this case.

```suggestion
			c.EXPECT().
				Get(gomock.Any(), kutil.Key(namespace, name), &extensionsv1alpha1.Infrastructure{}).
				Return(apierrors.NewNotFound(schema.GroupResource{}, name)).AnyTimes()
```